lib/mutant/ast/regexp/transformer/quantifier.rb

```diff
@@ -0,0 +1,97 @@
+module Mutant
+  module AST
+    module Regexp
+      class Transformer
+        class Quantifier < self
+          %i[
+            regexp_greedy_zero_or_more
+            regexp_greedy_one_or_more
+            regexp_greedy_zero_or_one
+            regexp_possessive_zero_or_one
+            regexp_reluctant_zero_or_more
+            regexp_reluctant_one_or_more
+            regexp_possessive_zero_or_more
+            regexp_possessive_one_or_more
+            regexp_greedy_interval
+            regexp_reluctant_interval
+            regexp_possessive_interval
+          ].map(&method(:register))
+
+          class ExpressionToAST < Transformer::ExpressionToAST
+            def call
+              ast(expression.min, expression.max)
+            end
+
+          private
+
+            def ast_type
+              :"regexp_#{expression.mode}_#{expression.token}"
+            end
+          end
+
+          class ASTToExpression < Transformer::ASTToExpression
+            include NamedChildren
+
+            children :min, :max, :subject
+
+            QUANTIFIER_MAP = IceNine.deep_freeze(
+              regexp_greedy_zero_or_more:     [:zero_or_more, '*',  :greedy],
+              regexp_greedy_one_or_more:      [:one_or_more,  '+',  :greedy],
+              regexp_greedy_zero_or_one:      [:zero_or_one,  '?',  :greedy],
+              regexp_possessive_zero_or_one:  [:zero_or_one,  '?+', :possessive],
+              regexp_reluctant_zero_or_more:  [:zero_or_more, '*?', :reluctant],
+              regexp_reluctant_one_or_more:   [:one_or_more,  '+?', :reluctant],
+              regexp_possessive_zero_or_more: [:zero_or_more, '*+', :possessive],
+              regexp_possessive_one_or_more:  [:one_or_more,  '++', :possessive],
+              regexp_greedy_interval:         [:interval,     '',   :greedy],
+              regexp_reluctant_interval:      [:interval,     '?',  :reluctant],
+              regexp_possessive_interval:     [:interval,     '+',  :possessive]
+            )
+
+            def call
+              expression = Regexp.to_expression(subject)
+
+              expression.quantify(type, text, min, max, mode)
+
+              expression
+            end
+
+          private
+
+            def text
+              if type.equal?(:interval)
+                interval_text + suffix
+              else
+                suffix
+              end
+            end
+
+            def type
+              QUANTIFIER_MAP.fetch(node.type).first
+            end
+
+            def suffix
+              QUANTIFIER_MAP.fetch(node.type).fetch(1)
+            end
+
+            def mode
+              QUANTIFIER_MAP.fetch(node.type).last
+            end
+
+            def interval_text
+              return "{#{min}}" if min.equal?(max)
```

Comment:
Also note that `Numeric#positive?` is only available in ruby 2.3.0+. I don't know offhand what versions of ruby mutant targets officially, but this might be something to keep in mind.